### PR TITLE
osu!catch - "Playable"

### PR DESCRIPTION
**Pull request by EriiYenn**
**Dear Peppy and Developers,**
I managed to get **osu!catch** at least to playable state.
There are some things undone, like bad **timer on fruits in end combo**, but I didn't find the right thing to repair yet.
Also, I removed some **unnecessary dashes** in code, just in case to make it clearer.

Many things have to be done and I'm trying to find **bugs** and repair them as much as possible!!

*Best regards,*
**EriiYenn**

### CODE REVIEW

File: osu.Game.Rulesets.Catch/CatchRuleset.cs [hunk 1]
```diff
@@ -41,9 +41,9 @@ public override IEnumerable<Mod> GetModsFor(ModType type)
                             Mods = new Mod[]
                             {
                                 new CatchModHalfTime(),
-                                new CatchModDaycore(),
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/CatchRuleset.cs]: we leave these intentionally

File: osu.Game.Rulesets.Catch/Scoring/CatchScoreProcessor.cs [hunk 2]
```diff
@@ -36,9 +36,12 @@ protected override void SimulateAutoplay(Beatmap<CatchHitObject> beatmap)
                 }
 
                 var fruit = obj as Fruit;
-
                 if (fruit != null)
                     AddJudgement(new CatchJudgement { Result = HitResult.Perfect });
+
+               var droplet= obj as Droplet;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/Scoring/CatchScoreProcessor.cs]: space before equals

File: osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs [hunk 3]
```diff
@@ -15,7 +15,7 @@ namespace osu.Game.Rulesets.Catch.UI
 {
     public class CatchPlayfield : ScrollingPlayfield
     {
-        public const float BASE_WIDTH = 512;
+        public const float BASE_WIDTH = 640;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs]: pretty sure this is wrong?

File: osu.Game.Rulesets.Catch/UI/CatchRulesetContainer.cs [hunk 4]
```diff
@@ -40,6 +40,10 @@ protected override DrawableHitObject<CatchHitObject> GetVisualRepresentation(Cat
             if (stream != null)
                 return new DrawableJuiceStream(stream);
 
+            var droplet = h as Droplet;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/CatchRulesetContainer.cs]: droplets are always nested though?

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs [hunk 5]
```diff
@@ -190,15 +190,15 @@ public void Add(DrawableHitObject fruit)
             /// <returns>Whether the catch is possible.</returns>
             public bool AttemptCatch(CatchHitObject fruit)
             {
-                const double relative_catcher_width = CATCHER_SIZE / 2;
+                const double relative_catcher_width = CATCHER_SIZE;
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: this is probably correct, but should be in its own comit or own PR

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs [hunk 6]
```diff
@@ -190,15 +190,15 @@ public void Add(DrawableHitObject fruit)
...
 
                 // this stuff wil disappear once we move fruit to non-relative coordinate space in the future.
                 var catchObjectPosition = fruit.X * CatchPlayfield.BASE_WIDTH;
                 var catcherPosition = Position.X * CatchPlayfield.BASE_WIDTH;
 
-                var validCatch =
-                    catchObjectPosition >= catcherPosition - relative_catcher_width / 2 &&
-                    catchObjectPosition <= catcherPosition + relative_catcher_width / 2;
+                var validCatch =                                                                    //best setup I found
```
Comment (peppy) [inline osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: not sure what these comments are trying to achieve

**Comment by peppy** [comment]
- Please do not add headers/formatting to your PR descriptions
- Please commit one commit (and probably one PR) per issue you fix.
- Please do not make up version numbers for components.

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs (hunk 6 above)
Comment (EriiYenn) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: These just for information, which combination works the best for valid catching.

File: osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs (hunk 3 above)
Comment (EriiYenn) [reply osu.Game.Rulesets.Catch/UI/CatchPlayfield.cs]: When I played osu!catch, some fruits went out of screen!
So I make the playground bigger

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs (hunk 6 above)
Comment (peppy) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: this is known and intentional. the sprite/catcher is currently sized wrongly.

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs (hunk 5 above)
Comment (EriiYenn) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: I transported the `/2` statement to `validCatch` variable to unite it

File: osu.Game.Rulesets.Catch/Scoring/CatchScoreProcessor.cs (hunk 2 above)
Comment (EriiYenn) [reply osu.Game.Rulesets.Catch/Scoring/CatchScoreProcessor.cs]: I'm sorry, my bad.

File: osu.Game.Rulesets.Catch/UI/CatcherArea.cs (hunk 6 above)
Comment (EriiYenn) [reply osu.Game.Rulesets.Catch/UI/CatcherArea.cs]: I agree, that's why I made this combination not equal

File: osu.Game.Rulesets.Catch/CatchRuleset.cs (hunk 1 above)
Comment (EriiYenn) [reply osu.Game.Rulesets.Catch/CatchRuleset.cs]: Understand and my apologies

**Comment by EriiYenn** [comment]
Should I close this PR and try to make it better and more professional?
Also I'll redo all my changes and fight with your values to make it work!

**Comment by peppy** [comment]
Honestly catch is in a very volatile state and I'm not sure these change will be of any help. I'd highly recommend you wait a couple of days until I push my banana (spinner) implementation, as that changes quite a bit of how catch works at the moment.

**Comment by EriiYenn** [comment]
Thank you for feedback, I love catch and want to contribute on making it perfect!!
I'll take a look on other issues.

**Comment by peppy** [comment]
Closing for now.
